pkg/controller/infrastructure/infraflow/shared/extras.go

```diff
@@ -71,17 +71,3 @@ func (w *waiter) Done() {
 }
 
 type contextKey struct{}
-
-// FromContext retrieves a waiter from the current context or returns nil if there is none.
-func FromContext(ctx context.Context) *waiter {
-	v := ctx.Value(contextKey{})
-	if v == nil {
-		return nil
-	}
-
-	w, ok := v.(*waiter)
-	if !ok {
-		return nil
-	}
-	return w
-}
```

Comment:
This shouldn't be removed because it will be part of the library anyway when we move it to the new repo.